spec/validations_spec.cr

```diff
@@ -25,7 +25,25 @@ struct ObjectToValidate
     invalidate(attr, "must not be baz") if val == "baz"
   end
 
-  validate x, size: (1..10), size_not_square_of: 3, custom_rule: true
+  validate x, size: (1..10), size_not_square_of: 3, custom_rule: true, if: @predicate
+
+  def validate
+    previous_def
+    invalidate("x", "must not be qux") if x == "qux"
+  end
+end
+
+record ObjectToValidateForUnlessPrecidate, x : String, predicate : Bool = true
+
+struct ObjectToValidateForUnlessPrecidate
+  include Validations
+  include CustomValidations
+
+  rule custom_rule do |attr, val|
+    invalidate(attr, "must not be baz") if val == "baz"
+  end
+
+  validate x, size: (1..10), size_not_square_of: 3, custom_rule: true, unless: @predicate
```

Comment:
Please change the spec to `unless: some_function_call` so we're sure it works nicely with the object context

Reply:
Switched! 

P.S: I hope I understood what you meant correctly 😅